test/unit/select_plan_test.lua

```diff
@@ -339,3 +339,203 @@ g.test_table_count = function()
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }} },
+        conditions = {{ '>=', 'id', 2 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_pk_field_sharding_key_from_le_sign_pk_condition_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }} },
+        conditions = {{ '<=', 'id', 2 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_pk_field_sharding_key_from_gt_sign_pk_condition_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }} },
+        conditions = {{ '<', 'id', 2 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_pk_field_sharding_key_from_lt_sign_pk_condition_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }} },
+        conditions = {{ '<', 'id', 2 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_field_sharding_key_from_its_non_unique_single_field_secondary_index_condition = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 5 }} },
+        conditions = {{ '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 42 },
+    },
+    extract_field_sharding_key_from_its_multiple_fields_secondary_index_condition = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }},
+        -- Expected values.
+        sharding_key = { 'John' },
+    },
+    extract_table_sharding_key_from_two_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'name', 'John' }, { '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 'John', 42 },
+    },
+    extract_table_sharding_key_from_two_index_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'full_name', { 'John',  'Doe' } }, { '==', 'age', 42 }},
+        -- Expected values.
+        sharding_key = { 'John', 42 },
+    },
+    extract_table_sharding_key_from_eq_an_ge_conditions_returns_nil = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 3 }, { fieldno = 5 }} },
+        conditions = {{ '==', 'name', 'John' }, { '>=', 'age', 42 }},
+        -- Expected values.
+        sharding_key = nil,
+    },
+    extract_table_sharding_key_from_partial_conditions = {
+        -- Input values.
+        sharding_index = { parts = {{ fieldno = 1 }, { fieldno = 4 }} },
+        conditions = {{ '==', 'full_name', { nil,  'Doe' } }, { '==', 'id', 1 }},
+        -- Expected values.
+        sharding_key = { 1, 'Doe' },
+    },
+    extract_table_sharding_key_from_partial_conditions_with_nil_and_non_nil_for_same_value_in_different_conditions = {
```

Comment:
Gosha, length of variable is exceeded of line's length limit:

```
$ echo "extract_table_sharding_key_from_partial_conditions_with_nil_and_non_nil_for_same_value_in_different_conditions" | wc -c
111
```


https://www.tarantool.io/en/doc/latest/dev_guide/c_style_guide/
> The limit on the length of lines is 80 columns and this is a strongly preferred limit. As for comments, the same limit of 80 columns is applied.

Reply:
Well, the config in `.luacheckrc` declares that `max_line_length = 120`. Moreover, the rest of `crud` code do not satisfy it: 12 lines in `sharding_key.lua`, 18 kines in `plan.lua` before this PR, etc.

Reply:
Shortened it

Reply:
And why are you referring C style guide for Lua code?